### DISCUSSION TITLE
When role is set with elementinternals

### DISCUSSION THREAD

**Discussion by herrKlein** [KonnorRogers/shadow-dom-testing-library#62] [General]
Should this work? 

```const button = screen.getByShadowRole('button');```

finding the element when its role is set with elementInternals?

```js
class MyButton extends HTMLElement {
  #internals;

  constructor() {
    super();
    this.#internals = this.attachInternals();
    this.#internals.role = 'button';
  }
}

customElements.define('my-button', MyButton);
```

```js
import { render, screen } from 'shadow-dom-testing-library';

test('it finds the custom element by role', () => {
  document.body.innerHTML = `<my-button>Click me</my-button>`;
  
  // The custom element has a role of 'button', so it can be found.
  const button = screen.getByShadowRole('button');
  expect(button).toBeInTheDocument();
});

```

**Comment by KonnorRogers** [comment]
does it work with regular `getByRole`? None of this would be shadow DOM specific, so if regular `getByRole` doesn't work, then it's not supported.

my best guess is it won't work because the underlying aria-query doesn't use the actual browser a11y tree, but instead makes its own similar a11y tree by traversing all the elements.

**Comment by herrKlein** [comment]
Thanks for your reply. It looks like its working, also with an internal role. 
But to be sure I have to doublecheck if I did not also set the role attribute. 
I am closing this for now. 
Thanks for this shadow dom testing library, I am using this extensively!
